ethereum/contracts/Splitter.sol

```diff
@@ -2,72 +2,27 @@ pragma solidity >= 0.5.7;
 pragma experimental ABIEncoderV2;
 
 contract Splitter {
-  struct User {
-    string name;
-    address account;
-  }
-  User[] users;
   mapping(address => bool) accounts;
   address owner;
 
   constructor() public {
     owner = msg.sender;
   }
 
-  function registerUser(string memory name) public {
-    require(!accounts[msg.sender], 'A given address is already registered');
-
-    User memory newUser = User({
-      name: name,
-      account: msg.sender
-    });
-    users.push(newUser);
-    accounts[msg.sender] = true;
-  }
-
-  event EtherSplit();
-
-  function splitEther() public payable {
-    require(accounts[msg.sender], 'A given address is not registered');
-    require(users.length > 2, 'Please wait until at least 3 users are registered');
-    require(msg.value > 0, 'A given value should be bigger than 0 (wei)');
-    require(msg.value / (users.length - 1) > 0, 'A given value is too small to be sent to others');
-
-    uint value = msg.value / (users.length - 1);
-
-    for (uint i = 0; i < users.length; i++) {
-      User storage user = users[i];
-      if (msg.sender != user.account) {
-        address payable recipient = address(uint160(user.account));
-        recipient.transfer(value);
-      }
-    }
+  event Transfer(address from, bool status, uint balance1, uint balance2);
 
-    emit EtherSplit();
-  }
-
-  function getAllUsers() public view returns (User[] memory) {
-    return users;
-  }
-
-  function deleteUser() public {
-    require(accounts[msg.sender], 'A given account does not exist');
-
-    accounts[msg.sender] = false;
-    for (uint i = 0; i < users.length; i++) {
-      if (msg.sender == users[i].account) {
-        delete users[i];
-      }
-    }
-  }
+  function sendEther(address[] memory receivers) public payable {
+    // One transfer call needs 2300 gas
+    require(msg.sender.balance + 4600 > msg.value, 'A sender should have enough balance');
+    require(receivers.length == 2, 'The number of receivers should be 2');
+    require(msg.value % 2 == 0, 'The value should be evenly divisible');
+    require(msg.value > 0, 'A given value should be bigger than 0');
+    require(msg.sender != receivers[0] && msg.sender != receivers[1], 'A sender should not be one of receivers');
 
-  // only for unit test
-  function deleteAllUsers() public {
-    require(msg.sender == owner, 'Delete all users can be run by owner only');
+    uint value = msg.value / 2;
+    address(uint160(receivers[0])).transfer(value);
+    address(uint160(receivers[1])).transfer(value);
```

Comment:
Big trouble. 

Sender A sends funds and the contract splits to B and C.  Given that transactions are atomic, what will happen if receiver C refuses to accept the funds? There are some important heuristics to avoid a DoS attack. Let's see if you can recognize the issue and find a way to address it.

Reply:
I realized that a receiver could be a contract as well so if it does not have any fallback function with payable then sending funds will be failed.

Plus sender will loses funds as well.

So add some process to avoid this thing and please let me know I am wrong 😄 

Also I found some interesting thing from [here](https://ethereum.stackexchange.com/questions/42512/transfer-send-error-cases) answered by you! 😃 

`transfer()` will revert automatically when the transaction failed.

Here is my question, if receiver2 failed then is the transaction not reverted?
```js
receiver1.transfer(value);
receiver2.transfer(value);
```